docs/cosign.md

```diff
@@ -0,0 +1,64 @@
+# Container Image Sign and Verify with cosign tool
+
+[cosign](https://github.com/sigstore/cosign) is tool that allows you to sign and verify container images with the
+public/private key pairs or without them by providing
+a [Keyless support](https://github.com/sigstore/cosign/blob/main/KEYLESS.md).
+
+Keyless uses ephemeral keys and certificates, which are signed automatically by
+the [fulcio](https://github.com/sigstore/fulcio) root CA. Signatures are stored in
+the [rekor](https://github.com/sigstore/rekor) transparency log, which automatically provides an attestation as to when
+the signature was created.
+
+You can enable container signing and verifying features with `push` and `pull` commands of `nerdctl` by using `cosign`
+under the hood with make use of flags `--sign` while pushing the container image, and `--verify` while pulling the
+container image.
+
+> * Ensure cosign executable in your `$PATH`.
+> * You can install cosign by following this page: https://docs.sigstore.dev/cosign/installation
+
+Prepare your environment:
+
+```shell
+# Create a sample Dockerfile
+$ cat <<EOF | tee Dockerfile.dummy
+FROM alpine:latest
```

Comment:
It should be noted that we don't validate `FROM` images here (and it's fine under an assumption that Docker Hub official images are trustable)

Reply:
thanks done.